### PR TITLE
[FSK-37]: Configure GraphQl Codegen

### PR DESCRIPTION
**Pull request by mojtabaesl**
## What?
Added [Graphql-Codegen](https://www.the-guild.dev/graphql/codegen) to auto-generate our hooks

## Why?
These changes would decrease redundant work in creating hooks and types.

So:
- Mainly we don't need to call `useQuery` and pass types
- We can write our Qraphql queries in a separate .grahql file

### CODE REVIEW

**Comment by mojtabaesl** [comment]
Please look at the script section in the package.json file
I'm not sure about `"dev:local"` name

File: src/App/App.stories.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import type { Meta, Story } from '@storybook/react';
 
-import { GetAdminsIdQuery } from '../operations/queries';
+import { GetAdminListDocument } from '../operations/__generated__/typesAndHooks';
```
Comment (ASafaeirad) [inline src/App/App.stories.tsx:3]: If we re-export it in 'operations,' modifying the structure will be less expensive.

**Review by ASafaeirad — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -20,7 +20,11 @@
     "pr": "./scripts/pr",
     "cy:component:run": "cypress run --component",
     "cy:e2e:run": "cypress run",
-    "cy:e2e:open": "cypress open"
+    "cy:e2e:open": "cypress open",
+    "test:e2e:run": "start-server-and-test dev http://127.0.0.1:5173 cy:e2e:run",
+    "test:e2e:open": "start-server-and-test dev http://127.0.0.1:5173 cy:open",
```
[the remark this answers is not in the document]
Comment (mojtabaesl) [reply package.json:25]: ok

File: .eslintignore [hunk 3]
```diff
@@ -25,3 +25,5 @@ dist-ssr
 
 .cspell
 .husky
+
+apollo.config.js
```
[the remark this answers is not in the document]
Comment (mojtabaesl) [reply .eslintignore:29]: My bad
I was testing something and I forgot to add it to `tsconfig.eslint.json`

File: README.md [hunk 4]
```diff
@@ -42,6 +42,10 @@ npm install
 npm run dev
 ```
 
+> If you're using [vscode][vscode] you can install [Apollo
+> Graphql][apollo-graphql-extension] extension to get features like autocomplete
+> in your graphql queries
```
[the remark this answers is not in the document]
Comment (mojtabaesl) [reply README.md:47]: Amirabbas has already done it

File: src/App/App.stories.tsx (hunk 1 above)
Comment (mojtabaesl) [reply src/App/App.stories.tsx:3]: I thought that it's a good idea to see __generated__ in the importing path
to know that these are generated files
but really we don't need to know about it so I'll re-export

**Review by ASafaeirad — APPROVED** [review]

**Pull request merged**